### DISCUSSION TITLE
Can I pull ravendb into my own container

### DISCUSSION THREAD

**Discussion by georgiosd** [ravendb/ravendb#15580] [Q&A]
Hey everyone

I'm composing a docker image with an ASP.NET core app, its React frontend and I want to add in Raven, so that I can fully control the environment - the app will be deployed to clients with varying technical skills and I would prefer that I only have to explain to them about this one image, rather than using compose, etc.

Is there a way to do this, perhaps with `COPY --from`?

The alternative would be to run embedded of course, but I'd like to first explore this way, which would give me the flexibility to start RavenDB inside the container or use an external instance with less effort.

**Comment by gregolsky** [comment]
If you want you can copy RavenDB server files from `/opt/RavenDB`
See https://github.com/ravendb/ravendb/blob/v5.4/docker/ravendb-ubuntu/Dockerfile.x64#L25

However that decreases maintainability and probably will make  configuration and management harder for you down the line. For containers a rule of a thumb is to run a single thing per container.

**Reply by georgiosd** [reply]
Fascinating, thanks. I generally would agree with you, but this particular deployment needs to work as closely to an MSI installer as possible :)
